permissions/rules/za.js

```diff
@@ -0,0 +1,129 @@
+const CanCan = require('cancan');
+const { isCounselorWhoCreated, isSupervisor, isCaseOpen } = require('./helpers');
+const Actions = require('../actions');
+const User = require('../user');
+const models = require('../../models');
+
+const cancan = new CanCan();
+const { can, allow } = cancan;
+
+const { Case } = models;
+
+allow(
+  User,
+  'edit',
```

Comment:
Would the approach be any simpler if instead of using `edit` for this argument all the time and having a callback, we instead put the action here, like `closeCase`?  Then this could look like:
```
  allow(User, 'closeCase', (user, caseObj) =>
    (isSupervisor(user) || (isCaseOpen(caseObj) && isCounselorWhoCreated(user, caseObj)))
  );
```
Might be a little more readable and concise.

Reply:
That's correct. I've misread **CanCan** lib and thought the second parameter should be one of: `view | create | edit | delete | manage`. But actually it can be any string or `manage`. Fixed it. (pushing soon)